mantidproject/mantid
# Add export button for the reduced workspace on the Preview tab

**Depends on PR #34186.** To review, view the comparison with [that branch](https://github.com/gemmaguest/mantid/compare/34106_perform_reduction_on_roi_changed...34106_export_button_for_reduced_ws)

This PR adds a toolbar button to export the reduced workspace on the reflectometry GUI Preview tab to the ADS.

**To test:**

This must be run in a development build.

Note that this is work in progress on a hidden GUI component so is very rough and ready at the moment. The look and usability of these features is currently very poor so you do not need to test that side thoroughly; just ensure it performs as expected in the happy path case and does not crash etc.

- Ensure this workspace is loaded or is in your load path: [INTER45455_inst.zip](https://github.com/mantidproject/mantid/files/7504266/INTER45455_inst.zip)
- Open the ISIS Reflectometry GUI
- Select the Preview tab.
- Type in `INTER45455_inst`, set the angle to `0.5`, and click Load. The instrument view plot should now display data. Click the rectangle-select button above it and select a region. The second plot will now be updated. 
- Click the rectangle-selection button on the second plot and select a region. A reduction will now be performed on the selected spectra and a log message `Reduction completed` should be displayed.
  - You can see the counts better on the second plot if you set the color scale to SymmetricLog10, although there's a bug with autoscaling (un-tick and re-tick to get it to work).
  - Note that the reduction is quite picky and may fail; if so you will get a warning in the log. It's best to select a region around spectra 20-30.
- Click the Export button _below_ the second plot. This should export the workspace `preview_reduced_ws` to the ADS. Double click it and you should get a 1D plot. View its history and it should have run ReflectometryReductionOneAuto with the settings on the GUI. These will be taken from the Experiment and Instrument settings tab apart from ProcessingInstructions and Angle which are taken from the Preview tab.

![image](https://user-images.githubusercontent.com/12895056/177527512-dc895fba-ebee-4ef3-99af-d5fb7f198e3a.png)

Part of #34106 

*This does not require release notes* because **it only affects functionality that is not yet available in the release build**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
